### DISCUSSION TITLE
Ideas for a "How Do I ..." document page

### DISCUSSION THREAD

**Discussion by gitbls** [gitbls/sdm#82] [General]
I've concluded that sdm users could benefit from a "How Do I ..." page. My first thought was that it would be a bulleted list of questions (e.g., How do I update fstab in my IMG with sdm?) and answers with links/pointers but I'm of course open to ideas on the basic structure as well.

Since I have a different perspective on sdm than everyone else, I'm interested in hearing your thoughts on topics that this page should cover. 

Thanks for taking the time to read and comment on this!

**Comment by stb68** [comment]
Not so much "how to" but, on the subject of documentation improvement (and maybe this will get other people thinking what would be good extra documentation to have like "how to")...

For the mkdir plugin, I was wondering if this plugin would make multiple directory levels in one go, as per the -p option to the shell mkdir command.  I was going to run a test to see, not that I need it(!), but I took a look at the plugin instead and can see that it does.

Would be nice if this was clear in the documentation and, when thinking about it, I then wondered if it performs the chown / chmod options on each created level?


I suppose a "How do I copy files and directories?" could also cover this (yes! back on the discussion topic after all).  Including using mkdir first, instead of relying on the mkdirif option, to get the copyfile target directory owner / permissions set as desired.

Maybe a special mention for /etc/skel when covering copydir and some options that could be passed to rsync (i.e. here you presumably want the files created with root as the owner rather than either the user you just created or the source owner).

**Reply by gitbls** [reply]
> Not so much "how to" but, on the subject of documentation improvement (and maybe this will get other people thinking what would be good extra documentation to have like "how to")...
> 
> For the mkdir plugin, I was wondering if this plugin would make multiple directory levels in one go, as per the -p option to the shell mkdir command. I was going to run a test to see, not that I need it(!), but I took a look at the plugin instead and can see that it does.
> 
Sorry for the inexcusably long delay in responding. The first point, yes, the mkdir plugin uses `-p`. Always. Will update the documentation.

> Would be nice if this was clear in the documentation and, when thinking about it, I then wondered if it performs the chown / chmod options on each created level?
> 
At the moment, however, the chown/chmod are only done on the last directory. Not clear to me that it should do this on every level, though. But if it doesn't, there's obviously a feature gap since there's no way to set these with any other plugin. Thinking about this...appreciate any/all thoughts on it.

> I suppose a "How do I copy files and directories?" could also cover this (yes! back on the discussion topic after all). Including using mkdir first, instead of relying on the mkdirif option, to get the copyfile target directory owner / permissions set as desired.
> 
I agree that "Copying files and directories"  doc coverage would be a good idea. Popping this on the "do someday" list.

> Maybe a special mention for /etc/skel when covering copydir and some options that could be passed to rsync (i.e. here you presumably want the files created with root as the owner rather than either the user you just created or the source owner).

Um, what does /etc/skel have to do with copydir and/or rsync? The etc/skel stuff is only referenced in the `user` plugin.

**Comment by stb68** [comment]
> At the moment, however, the chown/chmod are only done on the last directory. Not clear to me that it should do this on every level, though. But if it doesn't, there's obviously a feature gap since there's no way to set these with any other plugin. Thinking about this...appreciate any/all thoughts on it.

I'm not 100% sure on this myself, though if you were creating a directory structure in a user's home area I would think you'd probably want all levels updated.

> Um, what does /etc/skel have to do with copydir and/or rsync? The etc/skel stuff is only referenced in the user plugin

If I have a source directory with files that may not be owned by root then I would want them to be owned by root once I have had them copied to /etc/skel as that's how I believe they should be owned then.  (I also have an idea that it didn't copy the files on user creation if they weren't owned by root, or didn't chown them on user creation, but I may well be making that up, it's quite a while since I tried it!)

So, I'm just suggesting that as an example of showing the flags to use.
